packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,157 @@
+<script lang="ts" setup>
+import type { ButtonProps } from "./types";
+const props = withDefaults(defineProps<ButtonProps>(), {
+  label: "",
+  isLoading: false,
+  isDisabled: false,
+  type: "button",
+  color: "primary",
+  variant: "default",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the button is clicked
+   */
+  click: [];
+}>();
+</script>
+
+<template>
+  <div
+    tabindex="0"
+    class="onyx-button"
+    :class="{
+      [`onyx-button--${props.color}`]: true,
+      [`onyx-button--${props.variant}--${props.color}`]: true,
+    }"
+  >
+    <button :disabled="props.isDisabled" class="onyx-button__element" @click="emit('click')">
+      <span class="onyx-button__element__label">{{ props.label }}</span>
```

Comment:
```suggestion
      <span class="onyx-button__label">{{ props.label }}</span>
```

In BEM only one `__` is allowed as far as I know :)